### DISCUSSION TITLE
[Call for Volunteers] WeCom Puppets: wechaty-puppet-wxwork is going to ready for testing!

### DISCUSSION THREAD

**Discussion by windmemory** [wechaty/wechaty#2057] [Announcements]
Hello to developers who are using Wechaty,

We have excellent news for you: 

**We have launched a new puppet `wechaty-puppet-wxwork`. It is ready for testing, and Alpha Testing Volunteers are needed.**

`wechaty-puppet-wxwork` is a **WeCom(original WeChat Work) solution**. Currently it supports following features:

- Send / Receive Text, Image, Video, File, UrlLink, MiniProgram messages
- Get Corporation and Individual contacts
- Get All kinds of rooms with room members
- Modify alias, phone, corporation and description on a contact
- Add / Remove members to / from room
- Modify room topic
- Room events(join, leave, topic)

If you want to benefit from the new puppet, you are welcomed to apply for joining the Alpha Testing List: https://github.com/juzibot/wxwork-tester

You can leave a comment under this issue, introduce yourself about who you are and what you want to do (**please DO follow the following TEMPLATE and fill it in details, or you might not be able to be selected**), so that we can be able to add you to the alpha testing list if you are qualified.

---

## Wechaty Puppet Wxwork Alpha Tester Apply Template

1. Introduce Yourself(who you are):

2. Explain the Problem You Want to Solve:(what you do):

3. Are Your Bot Related with the Following Tasks? (YES/NO)
   - Gambling
   - Adult & Sex

4. Your Email:

5. Your Wechat[must, please make sure this WeChat is searchable]:

6. I Had Joined the Wechaty Developers' Home Wechat Room by Scan QRCode from README.md: (Please join the group if you are not, and then say YES at here)

7. I Had Read the Wechaty API Doc: http://wechaty.js.org/docs/introduction :(YES/NO)

8. I Had Used wechaty-puppet-padplus or wechaty-puppet-donut Before Taking Part in This Test: (YES/NO)

9. I understand that the Wechaty Puppet Wxwork is in very Early Alpha Stage, and I have the necessary engineering technics to deal with the bugs instead of just asking for support: (Please say YES, then followed by your Name at here)

Optional:

* [ ] Strong willingness & motivation in wechaty-puppet-wxwork
* [ ] Have the ability & engineering power to fix the basic bug instead of just asking for support
* [ ] Good to communicate with the developers in the community
* [ ] The basic requirement for bot online time: we might take back the test token back if the bot is offline for more than 1 day.
* [ ] Willing to write a blog to share the experience of using wechaty-puppet-wxwork

**Comment by lhr0909** [comment]
## Wechaty Puppet Wxwork Alpha Tester Apply Template
1. Introduce Yourself(who you are): Simon Liang, CEO of https://xanthous.cn
2. Explain the Problem You Want to Solve:(what you do): Automate WxWork groups management using bots, and integrate into https://github.com/xanthous-tech/chat-operator
3. Are Your Bot Related with the Following Tasks? (YES/NO)
  * Gambling - NO
  * Adult & Sex - NO
4. Your Email: simon@x-tech.io
5. Your Wechat[must, please make sure this WeChat is searchable]: 27326835
6. I Had Joined the Wechaty Developers' Home Wechat Room by Scan QRCode from README.md: (Please join the group if you are not, and then say YES at here) YES
7. I Had Read the Wechaty API Doc: http://wechaty.js.org/docs/introduction :(YES/NO) YES
8. I Had Used wechaty-puppet-padplus or wechaty-puppet-donut Before Taking Part in This Test: (YES/NO) YES
9. I understand that the Wechaty Puppet Wxwork is in very Early Alpha Stage, and I have the necessary engineering technics to deal with the bugs instead of just asking for support: (Please say YES, then followed by your Name at here)

YES, Simon Liang

Optional:

* [x]  Strong willingness & motivation in wechaty-puppet-wxwork
* [x]  Have the ability & engineering power to fix the basic bug instead of just asking for support
* [x]  Good to communicate with the developers in the community
* [x]  The basic requirement for bot online time: we might take back the test token back if the bot is offline for more than 1 day.
* [x]  Willing to write a blog to share the experience of using wechaty-puppet-wxwork

**Comment by su-chang** [comment]
@lhr0909 your application are approved!

**Comment by wyq09** [comment]
### **Wechaty Puppet Wxwork Alpha Tester Apply Template**
Introduce Yourself(who you are): wu yiqun, developer of http://www.xinxiaodian.com
Explain the Problem You Want to Solve:(what you do): Automate WxWork groups management using bots
Are Your Bot Related with the Following Tasks? (YES/NO)
Gambling - NO
Adult & Sex - NO
Your Email: 407069461@qq.com
Your Wechat[must, please make sure this WeChat is searchable]: wuyiqun_9
I Had Joined the Wechaty Developers' Home Wechat Room by Scan QRCode from README.md: (Please join the group if you are not, and then say YES at here) YES
I Had Read the Wechaty API Doc: http://wechaty.js.org/docs/introduction :(YES/NO) YES
I Had Used wechaty-puppet-padplus or wechaty-puppet-donut Before Taking Part in This Test: (YES/NO) YES
I understand that the Wechaty Puppet Wxwork is in very Early Alpha Stage, and I have the necessary engineering technics to deal with the bugs instead of just asking for support: (Please say YES, then followed by your Name at here)
YES, wu yiqun

Optional:

- [x]  Strong willingness & motivation in wechaty-puppet-wxwork
- [x] Have the ability & engineering power to fix the basic bug instead of just asking for support
- [x] Good to communicate with the developers in the community
- [x]The basic requirement for bot online time: we might take back the test token back if the bot is offline for more than 1 day.
- [x] Willing to write a blog to share the experience of using wechaty-puppet-wxwork

**Comment by huan** [comment]
I believe the testing period is closed because we are using https://github.com/wechaty/puppet-service-providers for our Wechaty Puppet Service Providers to accept the token applications.

Please go to https://github.com/wechaty/puppet-service-providers/issues/new/choose to apply a token that you need, and file an issue with the right token type to send the application.

Link to https://github.com/wechaty/puppet-service-providers/issues/1
